packages/libs/pactjs-generator/src/contract/parsing/pactParser.ts

```diff
@@ -0,0 +1,410 @@
+      name,
+    }));
+  });
+}
+type IsNotDuplicated = <T>(
+  isEqual: (a: T, b: T) => boolean,
+) => (a: T, idx: number, list: T[]) => boolean;
+
+const isNotDuplicated: IsNotDuplicated =
+  <T>(isEqual: (a: T, b: T) => boolean) =>
+  (a: T, idx: number, list: T[]): boolean =>
+    list.findIndex((b) => isEqual(a, b)) === idx;
+
+// eslint-disable-next-line @rushstack/typedef-var
+const isNotDuplicatedModule = isNotDuplicated<IModuleLike>(
+  (a, b) => a.name === b.name && a.namespace === b.namespace,
+);
+
+function fileParser(
+  contract: string,
+  namespace: string = '',
+): [IModule[], IPointer] {
+  const pointer = getPointer(contract);
+  const result = parser(pointer);
+  if (result === FAILED) {
+    console.error('there is no rule matched with this token', pointer.next());
+    throw new Error('parsing error');
+  }
+  const tree = unwrapData(result);
+  const { modules, namespaces, usedModules } = tree;
+
+  const extModules =
+    modules !== undefined
+      ? modules.map(({ location, module: mod }) => ({
+          ...mod,
+          namespace: getNamespace(location, namespaces) || namespace,
+          usedModules: [
+            ...(mod.usedModules || []),
+            ...getUsedModules(location, usedModules),
+            ...getUsedModulesInFunctions(mod.functions),
+          ].filter(isNotDuplicatedModule),
+        }))
+      : [];
+
+  return [extModules, pointer];
+}
+
+interface IModuleLoader {
+  getModule(moduleFullName: string): Promise<IModuleWithPointer | undefined>;
+  parserFile(content: string): void;
+  getStorage(): Map<string, IModuleWithPointer>;
+}
+
+const moduleLoader = (
+  fetchModule: (moduleFullName: string) => Promise<string>,
+): IModuleLoader => {
+  const storage = new Map<string, IModuleWithPointer>();
+
+  const parseModule = (content: string, namespace?: string): void => {
+    const [modules, pointer] = fileParser(content, namespace);
+    modules.forEach((mod) => {
+      storage.set(getModuleFullName(mod), { ...mod, pointer });
+    });
+  };
+
+  const cache = new Map<string, Promise<string>>();
+  const fetchContract = (name: string): Promise<string> => {
+    if (cache.has(name)) return cache.get(name)!;
+    const pr = fetchModule(name);
+    cache.set(name, pr);
+    return pr;
+  };
+
+  return {
+    async getModule(moduleFullName: string) {
+      const slogs = moduleFullName.split('.');
+      const namespace = slogs.length === 2 ? slogs[0] : '';
+
+      const mod = storage.get(moduleFullName);
+      if (mod !== undefined) return mod;
+
+      const content = await fetchContract(moduleFullName);
+      if (content) {
+        parseModule(content, namespace);
+      }
+
+      return storage.get(moduleFullName)!;
+    },
+    parserFile(content: string) {
+      parseModule(content);
+    },
+    getStorage() {
+      return storage;
+    },
+  };
+};
+
+interface IModuleWithPointer extends IModule {
+  pointer?: IPointer;
+}
+
+// parse a module and its dependencies
```

Comment:
If you want to add specific docs, you can add it by using `/** */`. That will be part of the generated reference docs as well. You can also add semver indicators. See also https://api-extractor.com/pages/overview/intro/ 
```suggestion
/**
 * parse a module and its dependencies
 * @alpha
 */
```